wgpu-hal/src/vulkan/device.rs

```diff
@@ -546,11 +546,29 @@ impl super::Device {
         } else {
             vk::ColorSpaceKHR::SRGB_NONLINEAR
         };
-        let info = vk::SwapchainCreateInfoKHR::builder()
-            .flags(vk::SwapchainCreateFlagsKHR::empty())
+
+        let original_format = self.shared.private_caps.map_texture_format(config.format);
+        let mut raw_flags = vk::SwapchainCreateFlagsKHR::empty();
+        let mut raw_view_formats: Vec<vk::Format> = vec![];
+        if !config.view_formats.is_empty()
+            && self
+                .enabled_device_extensions()
+                .contains(&vk::KhrSwapchainMutableFormatFn::name())
```

Comment:
This looks like we'd silently ignore view formats.
Considering the extension has almost 0 support on android, we don't want to require it either.

I think we need a new downlevel flag for this capability.

Reply:
Now there is a problem I don't know how to handle properly: we already have a `VIEW_FORMATS` downlevel flag, and inside the `frame.texture.create_view()` we no longer know if the texture is from `SurfaceTexture` or not.
https://github.com/gfx-rs/wgpu/blob/964c94a02dd13fdde4faf8ffb4a1b6756aa84fd7/wgpu-core/src/device/mod.rs#L1031-L1037

Reply:
I think as long as we require the new downlevel flag in the configure fn then we know it should not be possible to get a surface texture with view formats set on it.

Reply:
It looks like also need to move `require_downlevel_flags(VIEW_FORMATS)?` from `create_texture_view()` to `create_texture()`?

Reply:
Oh yeah, ideally we'd check that as soon as possible. @cwfitzgerald is there any reason why we have it in `create_texture_view()` right now?

Regarding the new downlevel flag name, maybe we should shorten it to just `SURFACE_VIEW_FORMATS`?

Reply:
OK, renamed.

Reply:
@teoxoy not that I know of - it should be texture creation